src/test/java/org/prebid/server/bidder/mgid/MgidBidderTest.java

```diff
@@ -0,0 +1,190 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.prebid.server.proto.openrtb.ext.response.BidType.banner;
+
+public class MgidBidderTest extends VertxTest {
+
+    private static final String ENDPOINT_URL = "https://test.endpoint.com/mgid-exchange/";
+
+    private MgidBidder mgidBidder;
+
+    @Before
+    public void setUp() {
+        mgidBidder = new MgidBidder(ENDPOINT_URL);
+    }
+
+	@Test
+    public void creationShouldFailOnInvalidEndpointUrl() {
+        assertThatIllegalArgumentException().isThrownBy(() -> new MgidBidder("invalid_url"));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorIfImpExtCouldNotBeParsed() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(Imp.builder()
+                        .ext(mapper.valueToTree(ExtPrebid.of(null, mapper.createArrayNode())))
+                        .build()))
+                .id("request_id")
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = mgidBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1);
+        assertThat(result.getErrors().get(0).getMessage()).startsWith("Cannot deserialize instance");
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldAddBidFloorCurAndBidFloorAndTmaxToIncomingRequest() {
+        // given
+        String currency = "GRP";
+        BigDecimal bidFlor = new BigDecimal(10.3);
+        String placementId = "placID";
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(Imp.builder()
+                        .ext(mapper.valueToTree(ExtPrebid.of(null,
+                                ExtImpMgid.of("accId", placementId, currency, currency,
+                                        bidFlor, bidFlor))))
+                        .build()))
+                .id("reqID")
+                .build();
+
+        BidRequest expected = BidRequest.builder()
+                .imp(singletonList(Imp.builder()
+                        .bidfloor(bidFlor)
+                        .bidfloorcur(currency)
+                        .tagid(placementId)
+                        .ext(mapper.valueToTree(ExtPrebid.of(null,
+                                ExtImpMgid.of("accId", placementId, currency, currency,
+                                        bidFlor, bidFlor))))
+                        .build()))
+                .tmax(200L)
+                .id("reqID")
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = mgidBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getValue()).hasSize(1)
+                .extracting(HttpRequest::getBody)
+                .containsOnly(Json.encode(expected));
+    }
+
```

Comment:
Please add some more specific test cases for when bidder throws its errors and for handling variables and values from impression extension. For example, the case in `getCur` method, when value is taken from another field.

We need to have all possible scenarios covered (i.e. any possible if or catch case, etc. )